pennylane/ops/op_math/sprod.py

```diff
@@ -137,8 +138,13 @@ def circuit(scalar, theta):
     def __init__(self, scalar: Union[int, float, complex], base: Operator, do_queue=True, id=None):
         super().__init__(base=base, scalar=scalar, do_queue=do_queue, id=id)
 
-        if base_pauli_rep := getattr(self.base, "_pauli_rep", None):
-            pr = {pw: qnp.dot(coeff, self.scalar) for pw, coeff in base_pauli_rep.items()}
+        if (base_pauli_rep := getattr(self.base, "_pauli_rep", None)) and (self.batch_size is None):
+            scalar = copy(self.scalar)
+            if qnp.get_interface(scalar) == "tensorflow" and not scalar.dtype.is_complex:
```

Comment:
grumble grumble tensorflow grumble grumble.

Reply:
Ya its really ugly :(

Reply:
One option, thought it might require some refactoring, is to add an interface / dtype attribute to the PS class. Then on construction, we always convert all of the parameters to vanilla numpy. After the computation, when a user wants a matrix or an operation, we cast it back to what it should be?